File: src/main/java/com/github/svegon/capi/mixininterface/IMinecraftClient.java
package com.github.svegon.capi.mixininterface;

import net.minecraft.client.render.RenderTickCounter;
import net.minecraft.client.search.SearchManager;
import net.minecraft.client.toast.TutorialToast;
import net.minecraft.client.util.WindowProvider;
import net.minecraft.util.profiler.TickTimeTracker;

public interface IMinecraftClient {
   WindowProvider getWindowProvider();

   RenderTickCounter getRenderTickCounter();

   void setRenderTickCounter(RenderTickCounter renderTickCounter);

   SearchManager getSearchManager();

   void setSearchManager(SearchManager searchManager);

   int getItemUseCooldown();

   void setItemUseCooldown(int cooldown);

   int getAttackCooldown();

   void setAttackCooldown(int cooldown);

   float getPausedTickDelta();

   long lastMetricsSampleTime();

   long nextDebugInfoUpdateTime();

   TutorialToast getSocialInteractionsToast();

   void setSocialInteractionsToast(TutorialToast toast);

   TickTimeTracker getTickTimeTracker();

   void attack();

   void pickItem();

   void useItem();

   void progressBlockBreaking();
}

File: src/main/kotlin/io/github/svegon/mclientapi/mixininterface/IMinecraftClient.kt
package io.github.svegon.capi.mixininterface

import net.minecraft.client.render.RenderTickCounter

interface IMinecraftClient {
    val windowProvider: WindowProvider?

    fun getRenderTickCounter(): RenderTickCounter?

    fun setRenderTickCounter(renderTickCounter: RenderTickCounter?)

    fun getSearchManager(): SearchManager?

    fun setSearchManager(searchManager: SearchManager?)

    var itemUseCooldown: Int

    var attackCooldown: Int

    val pausedTickDelta: Float

    fun lastMetricsSampleTime(): Long

    fun nextDebugInfoUpdateTime(): Long

    fun getSocialInteractionsToast(): TutorialToast?

    fun setSocialInteractionsToast(toast: TutorialToast?)

    val tickTimeTracker: TickTimeTracker?

    fun attack()

    fun pickItem()

    fun useItem()

    fun progressBlockBreaking()
}
